kernel/declarations.ml

```diff
@@ -100,9 +100,17 @@ type typing_flags = {
 
 }
 
+type work_list = (Univ.Instance.t * Id.t array) Cmap.t *
+  (Univ.Instance.t * Id.t array) Mindmap.t
+
+type cooking_info = {
+  modlist : work_list;
+  abstract : Constr.named_context * Univ.Instance.t * Univ.AUContext.t;
+}
+
 (* some contraints are in constant_constraints, some other may be in
  * the OpaqueDef *)
-type 'opaque constant_body = {
+type 'opaque constant_body_g = {
```

Comment:
I am not sure I like this naming scheme. We already use it for other purposes, and in the kernel we rather prefix parameteric types with `p`.

Reply:
@SkySkimmer this is the only issue that prevents me from merging this PR, I believe.

Reply:
>We already use it for other purposes

Like what?

Isn't the `p` stuff about universes? The only examples I remember are things like `pconstant` (not parametric) or `puniverses`.

Reply:
I think that `p` stands both for parametric or polymorphic. We have it in constr also for generic terms. But in any case, the `_g` form is used in the upper layers for CAst-wrapped ASTs, so it shouldn't be used in the kernel for something unrelated.